### PR TITLE
Add parser for table section

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: test/unittests/parser_test.cpp [hunk 1]
```diff
@@ -257,6 +257,48 @@ TEST(parser, function_section_with_multiple_functions)
     EXPECT_EQ(module.funcsec[3], 0xff);
 }
 
+TEST(parser, table_single_min_limit)
+{
+    const auto section_contents =
+        bytes{} + uint8_t{0x01} + uint8_t{0x70} + uint8_t{0x00} + uint8_t{0x7f};
```
Comment (axic) [inline test/unittests/parser_test.cpp]: The file uses `uint8_t{}` allover the place, I suggest it is fixed in a follow up PR. Currently these tests mirror the memory section tests (as they are very similar).

File: lib/fizzy/parser.cpp [hunk 2]
```diff
@@ -259,6 +277,9 @@ Module parse(bytes_view input)
         if (it != expected_end_pos)
             throw parser_error{"incorrect section " + std::to_string(static_cast<int>(id)) +
                                " size, difference: " + std::to_string(it - expected_end_pos)};
+
+        if (module.tablesec.size() > 1)
```
Comment (axic) [inline lib/fizzy/parser.cpp]: I have another set of changes which updates the parser for the memory section, etc.

File: test/unittests/parser_test.cpp (hunk 1 above)
Comment (chfast) [inline test/unittests/parser_test.cpp]:
```suggestion
        bytes{0x01, 0x70, 0x00, 0x7f};
```
Not sure what do you want to fix here exactly.

File: test/unittests/parser_test.cpp [hunk 3]
```diff
@@ -257,6 +257,48 @@ TEST(parser, function_section_with_multiple_functions)
...
+    const auto bin = bytes{wasm_prefix} + make_section(4, section_contents);
```
Comment (chfast) [inline test/unittests/parser_test.cpp]:
```suggestion
    const auto bin = wasm_prefix + make_section(4, section_contents);
```

File: test/unittests/parser_test.cpp (hunk 1 above)
Comment (axic) [reply test/unittests/parser_test.cpp]: I want to replace the entire style in one commit. Similar changes apply to half of the file.

File: lib/fizzy/types.hpp [hunk 4]
```diff
@@ -22,6 +22,8 @@ enum class ValType : uint8_t
 /// https://webassembly.github.io/spec/core/binary/types.html#result-types
 constexpr uint8_t BlockTypeEmpty = 0x40;
 
+constexpr uint8_t FuncRef = 0x70;
```
Comment (gumb0) [inline lib/fizzy/types.hpp]: Maybe add link to https://webassembly.github.io/spec/core/binary/types.html#table-types

File: test/unittests/parser_test.cpp (hunk 1 above)
Comment (axic) [reply test/unittests/parser_test.cpp]: In #80.

**Pull request merged**